packages/astro/src/vite-plugin-utils/index.ts

```diff
@@ -50,3 +50,12 @@ const postfixRE = /[?#].*$/s;
 export function cleanUrl(url: string): string {
 	return url.replace(postfixRE, '');
 }
+
+const specialQueriesRE = /(?:\?|&)(?:url|raw|direct)(?:&|$)/;
```

Comment:
It'd be great if Vite exposed a blessed way to do this, I've had this usecase before

Reply:
Yeah perhaps it could. It already exposes APIs like `isCSSRequest`